pkg/compiler/codegen.go

```diff
@@ -1158,7 +1159,8 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 				c.convertStruct(lit, true)
 				return nil
 			}
-			c.prog.Err = fmt.Errorf("'&' can be used only with struct literals")
+			pos := c.buildInfo.config.Fset.Position(n.Pos())
+			c.prog.Err = fmt.Errorf("%s: '&' can be used only with struct literals", pos)
```

Comment:
> Note: I didn't find test cases covering this error so I didn't add any new test.

You may construct your own test-case. Take as a reference any compiler unit-test, e.g. https://github.com/nspcc-dev/neo-go/blob/4d2b88dd9dbd3d1acaf8cc45c6a8909372648d00/pkg/compiler/if_test.go#L9

Use `evalWithError` to check the compilation error.